src/app/products/services/localstorage.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import Product from '../interfaces/product.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LocalStorageService {
+  public setData(items: Map<Product, number>): void {
+    localStorage.setItem('cart', JSON.stringify([...items]));
+  }
+
+  public getData(): Map<Product, number> {
+    if (localStorage.getItem('cart')) {
```

Comment:
Better to move `localStorage.getItem('cart')` to separate variable instead of writing it twice